cirq-core/cirq/transformers/target_gatesets/compilation_target_gateset.py

```diff
@@ -170,10 +170,24 @@ def decompose_to_target_gateset(self, op: 'cirq.Operation', moment_idx: int) ->
         old_2q_gate_count = sum(1 for o in ops.flatten_to_ops(old_optree) if len(o.qubits) == 2)
         new_2q_gate_count = sum(1 for o in ops.flatten_to_ops(new_optree) if len(o.qubits) == 2)
         switch_to_new = (
-            any(op not in self for op in ops.flatten_to_ops(old_optree))
+            any(
+                protocols.num_qubits(op) == 2 and op not in self
+                for op in ops.flatten_to_ops(old_optree)
+            )
```

Comment:
Can we do this part of the check before constructing `new_optree` so we don't waste time on it if it's not needed?

Reply:
We will have to construct the `new_optree` irrespective of whether this check returns a True or False. 

We will not switch to the `new_optree` only if both the checks are false (i.e. all 2q operations in `old_optree` are in `self` and `old_2q_gate_count <= new_2q_gate_count`.